www/pfSense-pkg-Lightsquid/files/usr/local/pkg/lightsquid.inc

```diff
@@ -328,12 +335,16 @@ ssl.honor-cipher-order	= "enable"
 ssl.cipher-list		= "AES128+EECDH:AES256+EECDH:AES128+EDH:AES256+EDH:AES128-SHA:AES256-SHA:!aNULL:!eNULL:!DSS"
 EOF;
 
-		// Configure IPv4 and IPv6 sockets
+		/*
+		 * IPv4 HTTPS bind must NOT be a socket; otherwise the retarded
+		 * lighttpd thing yet again binds to 80/HTTP no matter what.
+		 * You enable SSL-only IPv4 by not configuring bind for IPv4 at all.
+		 * IPv6 SSL must be configured as a socket
+		 */
 		$lighttpd_ls_conf .= <<<EOF
 
```

Comment:
Thanks for the constructive feedback.  I'll do what I can to address them in the next release of lighttpd, specifically the current need to repeat the ssl config for each socket.

Regarding the behavior of lighttpd to default bind to port 80, that is intended as a user-friendly part of a simple config.  The global scope binds to a socket.  If you don't specify one in `server.bind`, it is "\*".  If you do not want the default port 80, then specify `server.port`.
```
server.port		= "{$lightsquid_config['lighttpd_ls_port']}"
\$SERVER["socket"]	== "0.0.0.0:{$lightsquid_config['lighttpd_ls_port']}" { 
    {$lighttpd_ls_conf_sslcommon}
}
\$SERVER["socket"]	== "[::]:{$lightsquid_config['lighttpd_ls_port']}" {
    {$lighttpd_ls_conf_sslcommon}
}
```

Regarding dual-stack IPv6, there is plenty of vitriol on the internet about IPv6 and dual-stack.  Still, I'll look into the possibility of simplifying, and will attempt to come up with a simple option for a commonly-desired config, again, with the complication of attempting reasonable backwards compatibility with existing lighttpd configs.

FYI: In the meantime, I have been spending plenty of time (much more than 8 hours :)) trying to get package maintainers for various distributions to update lighttpd and to accept patches for older versions.  Older version of lighttpd in stable Debian/Ubuntu releases are missing more than a few patches which address more than a few reported issues, though not (yet) the issues you reported above.